src/middlewared/middlewared/schema.py

```diff
@@ -555,3 +555,35 @@ def nf(*args, **kwargs):
 
         return nf
     return wrap
+
+
+class Unix(Attribute):
+
+    def __init__(self, *args, **kwargs):
+        self.perms = kwargs.pop('perms', False)   # Placeholder for perms
+        super().__init__(*args, **kwargs)
+
+    def clean(self, value):
+        if value is None and not self.required:
+            return self.default
+        return value
+
+    def validate(self, value):
```

Comment:
String like `77` would be accepted for permissions while it is not what we want. Just validate it with regexp `[0-7]{3}`

Reply:
I told him to do that way :p. I think it would be fine to do this way, just need to normalize, returning {mode:o3} (i dont recall if thats the exact formatting).

Reply:
@william-gr turns out `chmod` accepts `77` as valid mode so let's accept it too. Question solved.